native/src/main/java/io/ballerina/stdlib/data/csvdata/utils/CsvConfig.java

```diff
@@ -0,0 +1,240 @@
+package io.ballerina.stdlib.data.csvdata.utils;
+
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.ABSENT_AS_NILABLE;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.ALLOW_DATA_PROJECTION;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.COMMENT_CHAR;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.CUSTOM_HEADERS;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.DELIMITER;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.ENABLE_CONSTRAINT_VALIDATION;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.ESCAPE_CHAR;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.HEADER;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.LINE_TERMINATOR;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.LOCALE;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.NIL_AS_OPTIONAL;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.NIL_VALUE;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.SKIP_LINES;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.STRING_CONVERSION;
+import static io.ballerina.stdlib.data.csvdata.utils.Constants.ConfigConstants.TEXT_ENCLOSURE;
+
+public class CsvConfig {
```

Comment:
Do we need to make these fields mutable after initialization otherwise I feel like it's better to make this a record

Reply:
Since `stringConversion` field needs to change while traversing, we cannot convert that into a map.